cogs/general.py

```diff
@@ -74,14 +77,46 @@ async def on_message(self, message: discord.Message):
                 await message.channel.send(f"{mention.display_name} is AFK: {msg}")
 
         # Pull requests and issues
-        links = [
-            f"https://github.com/Pycord-Development/pycord/pull/{text[2:]}"
-            for text in message.content.split()
-            if text.startswith("##") and len(text) > 2 and text[2:].isdigit()
-        ][:3]
+        links = list(set([f'https://github.com/Pycord-Development/pycord/pull/{index}' for index in PULL_HASH_REGEX.findall(message.content)]))
         if links:
             await message.reply("\n".join(links))
 
+    @Cog.listener()
+    async def on_message_edit(self, before: discord.Message, after: discord.Message):
+        github_url_format = "https://github.com/Pycord-Development/pycord/pull/{index}" # a template for shorter code
+        if before.author.bot:
+            return
+        before_pull_indexes = PULL_HASH_REGEX.findall(before.content)
+        after_pull_indexes = PULL_HASH_REGEX.findall(after.content)
+        if not before_pull_indexes:
+            # it is a normal message
+            return
+        if after_pull_indexes == before_pull_indexes:
+            # pull indexes have not been changed yet
+            return
+        created_at = before.created_at
+        async for message in before.channel.history(limit=5, after=created_at):
+            if message.author.id != self.bot.user.id:
+                continue
+            content = message.content
+            all_links = list(link[2] for link in GITHUB_URL_REGEX.findall(content))
+            # we will modify the message content
+            # so it matches the new  pull indexes
+            splitted = content.split()
+            new_list = []
+            for word in splitted:
+                for link in all_links:
+                    try:
+                        link = github_url_format.format(index=after_pull_indexes[before_pull_indexes.index(link)])
+                    except IndexError:
+                        # new pull index added
+                        new_list.append(link)
+                    else:
+                        if link not in new_list:
+                            new_list.append(link)
+            await message.edit(content='\n'.join(new_list))
+            return
+
```

Comment:
I really dont think we should do this, at least not yet
```suggestion
```

Reply:
I'll make it as a null docstring to leave it.

Reply:
or maybe I'll just remove it so we can do it better in another time.